### ISSUE TITLE
Question Answering Fine-tuning - Finding the most relevant (question, answer) for a given query

### ISSUE THREAD

**Issue by yonatanbitton** [google-research/bert#639]
In the SQuAD dataset, the questions are relating to a context, a paragraph that can be asked about.

I want to use BERT without paragraph context.

I would like to fine-tune BERT on K question and answer pairs, and then finding the most relevant answer for a given query (in the predict)

Similar to this article:
https://arxiv.org/abs/1905.02851

How can I do such a thing?

Thanks.

**Comment by ArthurRizar** [comment]
i think you mean classic paraphrase identify. 
BERT is a little difference,   BERT-sentence-pair is a end2end paraphrase identify,  
feed two sentence sent_a and sent_b, and then concat them be a input_ids

you can use BERT or Siamese model

**Comment by yonatanbitton** [comment]
What you're describing sounds like the MRPC task for sentence pair classification.
For text_a and text_b, if they are similar, their label is 1, else 0. 
But it's NOT the mission that I describe. 

The mission I describe is to say if an answer is correct for a question, if something like this exists.

Here they check q-Q similarity: https://github.com/hanxiao/bert-as-service
But they do it in a different way (Encode and dot product) but I don't understand it fully ...

**Comment by ArthurRizar** [comment]
it named paraphrase identify. may be you need this https://github.com/tlatkowski/multihead-siamese-nets/blob/master/models/cnn.py

**Comment by yonatanbitton** [comment]
First, thanks for the link. 
From what I see about the data fields in this model: 
id - the id of a training set question pair
qid1, qid2 - unique ids of each question (only available in train.csv)
question1, question2 - the full text of each question
is_duplicate - the target variable, set to 1 if question1 and question2 have essentially the same meaning, and 0 otherwise.

It's just like Pytorch's BertForSequenceClassification:
I feed two texts (text_a, text_b) and a label - 1 or 0, and say if they have same meaning. 

The mission I describe if to feed question and answer pair (q, a) and to say if the anwer is answering the question.

**Comment by sandeshbsuvarna** [comment]
Does this help?

https://github.com/hanxiao/bert-as-service#building-a-qa-semantic-search-engine-in-3-minutes

**Comment by rpoli40** [comment]
@yonatanbitton Did you find an answer to your question?
